test/divergence_check_tests.cc

```diff
@@ -0,0 +1,205 @@
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/generators/catch_generators.hpp>
+#include <catch2/matchers/catch_matchers_string.hpp>
+
+#include <celerity.h>
+
+#include "divergence_check_test_utils.h"
+#include "log_test_utils.h"
+#include "test_utils.h"
+
+
+using namespace celerity;
+using namespace celerity::detail;
+using namespace celerity::test_utils;
+using celerity::access::fixed;
+
+TEST_CASE("test diverged task execution on device tasks", "[divergence]") {
+	using namespace cl::sycl::access;
```

Comment:
```suggestion
```
I realize you probably got this from some other test case, but this is very outdated. We have our own `celerity::access_mode` now!